src/content/cookbook/architecture/command.md

```diff
@@ -0,0 +1,616 @@
+
+  bool get running => // ...
+
+  Exception? get error => // ...
+
+  void load() {
+    // load user
+  }
+  // ···
+}
+```
+
+You can use the running state to display a progress indicator in the View:
+
+<?code-excerpt "lib/no_command.dart (ListenableBuilder)" replace="/\.load//g;/body: //g;/^\),$/)/g"?>
+```dart
+ListenableBuilder(
+  listenable: widget.viewModel,
+  builder: (context, _) {
+    if (widget.viewModel.running) {
+      return const Center(
+        child: CircularProgressIndicator(),
+      );
+    }
+    // ···
+  },
+)
+```
+
+Or use the running state to avoid executing the action multiple times:
+
+<?code-excerpt "lib/no_command.dart (load2)" replace="/2//g"?>
+```dart
+void load() {
+  if (running) {
+    return;
+  }
+  // load user
+}
+
+```
+
+Managing the state of an action can get complicated 
+if the ViewModel contains multiple actions. 
+For example, adding an `edit()` action to the `HomeViewModel` 
+can lead the following outcome:
+
+<?code-excerpt "lib/no_command.dart (HomeViewModel3)" replace="/(null|false);/\/\/ .../g;/3//g"?>
+```dart
+class HomeViewModel extends ChangeNotifier {
+  User? get user => // ...
+
+  bool get runningLoad => // ...
+
+  Exception? get errorLoad => // ...
+
+  bool get runningEdit => // ...
+
+  Exception? get errorEdit => // ...
+
+  void load() {
+    // load user
+  }
+
+  void edit(String name) {
+    // edit user
+  }
+}
+```
+
+Sharing the running state 
+between the `load()` and `edit()` actions might not always work, 
+because you might want to show a different UI component 
+when the `load()` action runs than when the `edit()` action runs,
+and you'll have the same problem with the `error` state.
+
+### Triggering UI actions from ViewModels
+
+Another challenge with ViewModel classes 
+is executing UI actions 
+when the ViewModel state changes. 
+
+For example, you might want to show a `SnackBar` when an error occurs, 
+or navigate to a different screen when an action completes. To implement this, listen for changes in the ViewModel, 
+and perform the action depending on the state. 
+
+In the View:
+
+<?code-excerpt "lib/no_command.dart (addListener)"?>
+```dart
+@override
+void initState() {
+  super.initState();
+  widget.viewModel.addListener(_onViewModelChanged);
+}
+
+@override
+void dispose() {
+  widget.viewModel.removeListener(_onViewModelChanged);
+  super.dispose();
+}
+```
+
+<?code-excerpt "lib/no_command.dart (_onViewModelChanged)" remove="widget.viewModel.clearError();"?>
+```dart
+void _onViewModelChanged() {
+  if (widget.viewModel.error != null) {
+    // Show Snackbar
+  }
+}
+```
+
+You need to clear the error state each time you execute this action, 
+otherwise this action happens each time `notifyListeners()` is called.
+
+<?code-excerpt "lib/no_command.dart (_onViewModelChanged)"?>
+```dart
+void _onViewModelChanged() {
+  if (widget.viewModel.error != null) {
+    widget.viewModel.clearError();
+    // Show Snackbar
+  }
+}
+```
+
+## Command pattern
+
```

Comment:
I see what you mean. Would it make sense to rewrite this recipe in a way that the command is introduced at the start? My idea was to show some of the issues when implementing view models and then fix them with the command pattern, but it can be simplified to just explaining the command pattern and how to use it.